### ISSUE TITLE
Support pretty printing as an option when writing Ion text

### ISSUE THREAD

**Issue by tgregg** [amzn/ion-python#53]

**Comment by tgregg** [comment]
Yes, that looks pretty good. I think it would be nice to have some way to keep small containers on a single line. For example,
`[4]`
is arguably more readable than
```
[
  4
]
```

**Comment by joshterrell805** [comment]
Agreed that `[4]` would look better. Hrmmm. I'll see if I can do that simply.

The approaches I can think of require buffering some of the stream/generator (perhaps the first two or three elements) until I can determine whether a container is "small" or not. Is it acceptable to buffer some of the ion_event stream instead of yielding all data for a given ion_event immediately?

**Comment by tgregg** [comment]
Yeah thinking more about it, I don't think it's worth the extra complexity at this point. As you point out, we'd probably have to add buffering and modify the coroutine transactions to get it to work. I'd suggest implementing the simpler version you outlined above; we can revisit some of the more complex options like this if there's sufficient demand.
